File: C#Integration/PokerProtocol/Commands/AbstractCommandResponse.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using EricUtility;

namespace PokerProtocol.Commands
{
    public abstract class AbstractCommandResponse<T>: AbstractCommand
        where T : AbstractCommand
    {
        private readonly T m_Command;

        public AbstractCommandResponse(T command)
        {
            m_Command = command;
        }
        protected T Command
        {
            get { return m_Command; }
        } 


        public override void Encode(StringBuilder sb) 
        {
            m_Command.Encode(sb);
        }

    }
}

File: JAVA/PokerProtocol/src/bluffinmuffin/protocol/commands2/AbstractCommandResponse.java
package bluffinmuffin.protocol.commands2;

public abstract class AbstractCommandResponse<T extends AbstractCommand> extends AbstractCommand
{
    private final T m_command;
    
    public AbstractCommandResponse(T command)
    {
        m_command = command;
    }
    
    public T getCommand()
    {
        return m_command;
    }
    
    @Override
    public void encode(StringBuilder sb)
    {
        m_command.encode(sb);
    }
}
